trafficgen-server-start

```diff
@@ -104,65 +122,46 @@ if [ "$switch_type" == "testpmd" ]; then
     if [ -e "$file" ]; then
         echo "Found $file:"
         cat $file
-        peermac0=`jq -r '.macs[0]' $file`
-        if [ ! -z "$peermac0" ]; then
-            echo "Found eth-peer MAC0 $peermac0"
-        fi
-        peermac1=`jq -r '.macs[1]' $file`
-        if [ ! -z "$peermac1" ]; then
-            echo "Found eth-peer MAC1 $peermac1"
-        fi
-    fi
-    if [ ! -z "$testpmd_dst_macs" ]; then
-        peermac0=`echo $testpmd_dst_macs | awk -F, '{print $1}'`
-        peermac1=`echo $testpmd_dst_macs | awk -F, '{print $2}'`
+        testpmd_dst_macs=`jq -r '.macs[]' $file | tr "\n" "," | sed -e 's/,$//'`
     fi
 
     # Build testpmd cmdline opts for device selection
     echo "Resolving devices for testpmd based on devices [$devices]"
     testpmd_devices=""
-    dev_1=$(echo "${devices}" | cut -d, -f 1)
-    dev_2=$(echo "${devices}" | cut -d, -f 2)
-    dev_3=$(echo "${devices}" | cut -d, -f 3-)
-    if [ -n "${dev_3}" ]; then
-        exit_error "Testpmd requires exactly 2 devices (not '${devices}'), exiting" 1 "$sample_dir"
-    fi
-    res_dev_1=""
-    resolve_device res_dev_1 "${dev_1}"
-    res_dev_2=""
-    resolve_device res_dev_2 "${dev_2}"
-    testpmd_devices+=" --allow ${res_dev_1}"
-    if [ -n "${testpmd_devopt}" ]; then
-        testpmd_devices+=",${testpmd_devopt}"
-    fi
-    testpmd_devices+=" --allow ${res_dev_2}"
-    if [ -n "${testpmd_devopt}" ]; then
-        testpmd_devices+=",${testpmd_devopt}"
-    fi
-    if [ -z "${testpmd_devices}" ]; then
-        exit_error "Testpmd devices could not be found, exiting" 1 "$sample_dir"
-    fi
-    res_dev_1_numa_node=$(cat "/sys/bus/pci/devices/${res_dev_1}/numa_node")
-    res_dev_2_numa_node=$(cat "/sys/bus/pci/devices/${res_dev_2}/numa_node")
-    echo "Device 1: ${res_dev_1}"
-    echo "Device 1 NUMA node: ${res_dev_1_numa_node}"
-    echo "Device 2: ${res_dev_2}"
-    echo "Device 2 NUMA node: ${res_dev_2_numa_node}"
 
-    if [ "${res_dev_1_numa_node}" == "-1" ]; then
-        res_dev_1_numa_node="0"
-    fi
+    count=0
+    for this_device in ${devices[@]}; do
+        resolve_device this_res_dev $this_device
+        res_devs[$count]=$this_res_dev
+        this_node=$(cat "/sys/bus/pci/devices/${res_devs[$count]}/numa_node")
+        if [ "$this_node" == "-1" ]; then
+            this_node=0
+        fi
```

Comment:
I think this is a totally fine assumption to make, but I do think we should log when we do this because it might be useful to know that we did when troubleshooting.

Reply:
OK, I'll add a message